src/Core/Services/EnvironmentService.cs

```diff
@@ -54,38 +56,50 @@ public string GetWebSendUrl()
             return GetWebVaultUrl(true) is string webVaultUrl ? $"{webVaultUrl}/#/send/" : DEFAULT_WEB_SEND_URL;
         }
 
+        public string GetCurrentDomain()
+        {
+            var url = WebVaultUrl ?? BaseUrl ?? ApiUrl ?? IdentityUrl;
+            if (!string.IsNullOrWhiteSpace(url))
+            {
+                if (url.Contains(Region.US.Domain()) || url.Contains(Region.EU.Domain()))
+                {
+                    return CoreHelpers.GetDomain(url);
+                }
+                return CoreHelpers.GetHostname(url);
+            }
+            return string.Empty;
+        }
```

Comment:
🤔 Can this be reused somehow? I see the same logic applied in `AccountView.ParseEndpoint(...)`

Reply:
Moved logic into EnvironmentUrlData.